plaso/storage/file_interface.py

```diff
@@ -0,0 +1,1147 @@
+      self._serialized_attribute_containers[container_type] = container_list
+
+    return container_list
+
+  def _SerializeAttributeContainer(self, attribute_container):
+    """Serializes an attribute container.
+
+    Args:
+      attribute_container (AttributeContainer): attribute container.
+
+    Returns:
+      bytes: serialized attribute container.
+
+    Raises:
+      IOError: if the attribute container cannot be serialized.
+      OSError: if the attribute container cannot be serialized.
+    """
+    if self._serializers_profiler:
+      self._serializers_profiler.StartTiming(
+          attribute_container.CONTAINER_TYPE)
+
+    try:
+      attribute_container_data = self._serializer.WriteSerialized(
+          attribute_container)
+      if not attribute_container_data:
+        raise IOError(
+            'Unable to serialize attribute container: {0:s}.'.format(
+                attribute_container.CONTAINER_TYPE))
+
+      attribute_container_data = attribute_container_data.encode('utf-8')
+
+    finally:
+      if self._serializers_profiler:
+        self._serializers_profiler.StopTiming(
+            attribute_container.CONTAINER_TYPE)
+
+    return attribute_container_data
+
+  def _RaiseIfNotReadable(self):
+    """Raises if the storage file is not readable.
+
+     Raises:
+      IOError: when the storage file is closed.
+      OSError: when the storage file is closed.
+    """
+    if not self._is_open:
+      raise IOError('Unable to read from closed storage file.')
+
+  def _RaiseIfNotWritable(self):
+    """Raises if the storage file is not writable.
+
+    Raises:
+      IOError: when the storage file is closed or read-only.
+      OSError: when the storage file is closed or read-only.
+    """
+    if not self._is_open:
+      raise IOError('Unable to write to closed storage file.')
+
+    if self._read_only:
+      raise IOError('Unable to write to read-only storage file.')
+
+class StorageFileReader(interface.StorageReader):
+  """File-based storage reader interface."""
+
+  def __init__(self, path):
+    """Initializes a storage reader.
+
+    Args:
+      path (str): path to the input file.
+    """
+    super(StorageFileReader, self).__init__()
+    self._path = path
+    self._storage_file = None
+
+  def GetFormatVersion(self):
+    """Retrieves the format version of the underlying storage file.
+
+    Returns:
+      int: the format version, or None if not available.
+    """
+    if self._storage_file:
+      return self._storage_file.format_version
+
+    return None
+
+  def GetSerializationFormat(self):
+    """Retrieves the serialization format of the underlying storage file.
+
+    Returns:
+      str: the serialization format, or None if not available.
+    """
+    if self._storage_file:
+      return self._storage_file.serialization_format
+
+    return None
+
+  def GetStorageType(self):
+    """Retrieves the storage type of the underlying storage file.
```

Comment:
Why is this relevant to expose? The idea of the reader is to hide the fact the storage is a file or something else

Reply:
This is only used by pinfo, to display information about the storage details of the store.

Reply:
ack, per conversation